specification/security/get_builtin_privileges/SecurityGetBuiltinPrivilegesResponse.ts

```diff
@@ -18,7 +18,8 @@
  */
 
 import { IndexName } from '@_types/common'
+import {ClusterPrivilege, ClusterPrivilegeForRemote} from "@security/_types/Privileges";
 
 export class Response {
-  body: { cluster: string[]; index: IndexName[] }
+  body: { cluster: ClusterPrivilege[]; index: IndexName[]; remote_cluster: ClusterPrivilegeForRemote[] }
```

Comment:
I am unsure about changing cluster from string[] to the proper enumeration

This should be the preferred model, where we guide users to only add named privileges from the enum.  However, it is technically possible (but not recommended) to add the action names directly which are just string[]. Other places use the enum and this makes them the same (and the validation still passes).

Reply:
This one is the response class. Are we sure that only valid privileges from the enum are returned here? If yes, we should definitely prefer the enum (however, this will be a breaking change for all statically typed language clients, so we have to check with everybody else f irst).

Reply:
From this API, yes, only values from the enum are ever returned.

Reply:
For me it would be ok to introduce a breaking change by changing the return type to the correct enum type. What about the rest @elastic/devtools-team ?

Reply:
See https://github.com/elastic/elasticsearch-specification/pull/3125#discussion_r1842003226 - turns out it's non breaking after all.